### PR TITLE
Support http-basic-auth via basicAuth setting

### PR DESCRIPTION
**Pull request by staabm**
use settings.basicAuth.username and settings.basicAuth.password to
configure basic authentication.

### CODE REVIEW

**Comment by staabm** [comment]
see https://github.com/scottgonzalez/node-wordpress/pull/36#discussion_r70427197

File: lib/wordpress.js [hunk 1]
```diff
@@ -44,7 +44,11 @@ function Client( settings ) {
 		// via settings.host
 		headers: {
 			Host: parsedUrl.host
-		}
+		},
+		basic_auth: settings.basicAuth ? {
+			user: settings.basicAuth.username,
+			pass: settings.basicAuth.password
+		} : null,
```
Comment (scottgonzalez) [inline lib/wordpress.js]: Please either break before the operators:

```js
		basic_auth: settings.basicAuth
			? {
				user: settings.basicAuth.username,
				pass: settings.basicAuth.password
			}
			: null
```

Or invert the conditional:

```js
		basic_auth: !settings.basicAuth ? null : {
			user: settings.basicAuth.username,
			pass: settings.basicAuth.password
		}
```

And drop the trailing comma.

**Comment by scottgonzalez** [comment]
Thanks, this still needs docs.

**Comment by staabm** [comment]
@scottgonzalez  documented the new options and adjusted CS per review

**Pull request closed without merging**